src/ws/login.js

```diff
@@ -74,21 +74,56 @@
         return document.getElementById(name);
     }
 
+    // Hide an element (or set of elements) based on a boolean
+    // true: element is hidden, false: element is shown
+    function hideToggle(elements, toggle) {
+        var els;
+
+        // If it's a single selector, convert it to an array for the loop
+        if (typeof elements === "string")
+            elements = [elements];
+
+        // >= 1 arguments (of type element or string (for CSS selectors))
+        // (passed in "arguments" isn't a a true array, so forEach wouldn't always work)
+        for (var i=0; i < elements.length; i++) {
+            if (typeof elements[i] === "string") {
+                // Support CSS selectors as a string
+                els = document.querySelectorAll(elements[i]);
+
+                if (els)
+                    els.forEach(function(element){
+                        element.hidden = !!toggle;
+                    });
+            } else {
+                // Hide specific elements
+                elements[i].hidden = !!toggle;
+            }
+        }
+    }
+
+    // Show >=1 arguments (element or CSS selector)
+    function show() {
+        hideToggle(arguments, false);
+    }
+
+    // Hide >=1 arguments (element or CSS selector)
+    function hide() {
+        hideToggle(arguments, true);
+    }
+
     function fatal(msg) {
         if (window.console)
             console.warn("fatal:", msg);
 
-        id("login-again").style.display = "none";
-        id("login-wait-validating").style.display = "none";
+        hide("#login-again", "#login-wait-validating");
 
         if (oauth_redirect_to) {
             id("login-again").href = oauth_redirect_to;
-            id("login-again").style.display = "block";
+            show("#login-again");
```

Comment:
`#login-again` is a `a` element, and giving it `display: block` had an effect. I think we should keep it `display: block` somehow, like with a explicit rule in login.css.

Reply:
See https://github.com/cockpit-project/cockpit/pull/14870#issuecomment-723864991 and https://github.com/cockpit-project/cockpit/pull/14870#issuecomment-723878719

Reply:
Yep. The display types should be in CSS, not JavaScript. I'll add it to the CSS.